app/models/work_activity.rb

```diff
@@ -68,9 +68,14 @@ def comment_html
     def file_changes_html
       changes = JSON.parse(message)
       changes_html = changes.map do |change|
-        "<i>#{change['action']}</i> <span>#{change['filename']}<span><br/>"
+        icon = if change["action"] == "deleted"
+                 '<i class="bi bi-file-earmark-minus-fill file-deleted-icon"></i>'
+               else
+                 '<i class="bi bi-file-earmark-plus-fill file-added-icon"></i>'
+               end
+        "<tr><td>#{icon}</td><td>#{change['action']}</td> <td>#{change['filename']}</td>"
```

Comment:
In ERB, by default it HTML-encodes templated values... but that's not happening here. We wouldn't expect to have any special characters in `change['action']` or `change['filename']`... but if there were user supplied data, we'd have a XSS vulnerability ... and it's hard to evaluate whether each case is safe or not.

One possibility would be to import https://api.rubyonrails.org/classes/ERB/Util.html#method-c-h and have a practice of using it any time we're interpolating plain strings into html

```suggestion
        "<tr><td>#{icon}</td><td>#{h(change['action'])}</td> <td>#{h(change['filename'])}</td>"
```

A complementary practice might be to mark html strings as such, so every interpolation is either `html_...` or `h(...)`

```suggestion
        "<tr><td>#{html_icon}</td><td>#{change['action']}</td> <td>#{change['filename']}</td>"
```

Or a different approach would be to use `html_safe`... but I think that could get confusing outside of ERB.

---------

But it probably makes sense to decide on a policy and apply it across the board, rather than worry about it here. But wanted to bring it up.

Reply:
These are all good observations. In this case the `change['action']` values are not user entered so we are OK. This is an example of how these values are added (see line 496): https://github.com/pulibrary/pdc_describe/blob/main/app/models/work.rb#L488-L498